pkg/provider/provider.go

```diff
@@ -126,18 +126,18 @@ func Provider() *schema.Provider {
 	}
 }
 
-func connectionString(host, username, password string, port int, database string, testing bool, application string) string {
+func connectionString(host, username, password string, port int, database string, sslmode bool, application string) string {
 	c := strings.Builder{}
 	c.WriteString(fmt.Sprintf("postgres://%s:%s@%s:%d/%s", username, password, host, port, database))
 
 	params := []string{}
 
-	if testing {
+	if sslmode {
+		params = append(params, "sslmode=require")
+	} else {
```

Comment:
Just realizing that we are setting feature flags twice, so we can probably remove
```
		params = append(params, "enable_rbac_checks=true")
		params = append(params, "enable_ld_rbac_checks=true")
```
Since all feature flags are set further down. Bigger discussion point I think this should still be called `testing` or something like that rather than `mz_sslmode` since setting it disables ssl and sets a number of specific feature flags.

Or maybe we break this out and have two parameters (hidden from users) that are `mz_sslmode` and `mz_preview_features` which is used to enable the specific feature flags.

Reply:
I like the idea of breaking this into two parameters! Will work on this today